opm/autodiff/MultisegmentWell.hpp

```diff
@@ -0,0 +1,357 @@
+/*
+  Copyright 2017 SINTEF Digital, Mathematics and Cybernetics.
+  Copyright 2017 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef OPM_MULTISEGMENTWELL_HEADER_INCLUDED
+#define OPM_MULTISEGMENTWELL_HEADER_INCLUDED
+
+
+#include <opm/autodiff/WellInterface.hpp>
+
+namespace Opm
+{
+
+    template<typename TypeTag>
+    class MultisegmentWell: public WellInterface<TypeTag>
+    {
+    public:
+        typedef WellInterface<TypeTag> Base;
+
+        // TODO: the WellState does not have any information related to segments
+        using typename Base::WellState;
+        using typename Base::Simulator;
+        using typename Base::IntensiveQuantities;
+        using typename Base::FluidSystem;
+        using typename Base::ModelParameters;
+        using typename Base::MaterialLaw;
+        using typename Base::BlackoilIndices;
+
+        /// the number of reservior equations
+        using Base::numEq;
+
+        using Base::has_solvent;
+        using Base::has_polymer;
+
+        // TODO: for now, not considering the polymer, solvent and so on to simplify the development process.
```

Comment:
Some of these comments look like they no longer apply, please remove any that are no longer useful.